datemath.go

```diff
@@ -151,6 +155,17 @@ func WithLocation(l *time.Location) func(*Options) {
 	}
 }
 
+// WithBusinessDayFn use the given fn to check if a day is a business day
+func WithBusinessDayFn(fn func(time.Time) bool) func(*Options) {
```

Comment:
Could we call this `WithBusinessDayFunc` instead? I think that is more common in the standard library, at least (e.g. `filepath.WalkFunc` and `strings.LastIndexFunc`). Same with the field on `Options`.

Reply:
Sounds good. Done.